src/com/puppetlabs/puppetdb/scf/hash.clj

```diff
@@ -132,16 +132,16 @@
   [{:keys [resource-type resource-title property timestamp status old-value
            new-value message file line] :as resource-event}]
   (generic-identity-string
-    { :resource-type resource-type
-      :resource-title resource-title
-      :property property
-      :timestamp timestamp
-      :status status
-      :old-value old-value
-      :new-value new-value
-      :message message
-      :file file
-      :line line}))
+    {:resource_type resource-type
```

Comment:
@senior we need to hash underscores because master uses underscores. This function is only used in hash generation though, so as long as we don't start using it elsewhere in stable I think we should be good.